File: app/src/main/java/com/softaai/designpatterns/java/creational/singleton/SingletonObjectDemo.java
package com.softaai.designpatterns.java.creational.singleton;

/**
 * Created by Amol Pawar on 29-03-2019.
 * softAai Apps
 */
public class SingletonObjectDemo {

    public static void main(String[] args){

        SingletonObject singletonObject = SingletonObject.getInstance();
        singletonObject.showMessage();

    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/singleton/SingletonObjectDemo.kt
package com.softaai.designpatterns.kotlin.creational.singleton

/**
 * Created by Amol Pawar on 29-03-2019.
 * softAai Apps
 */
object SingletonObjectDemo {

    @JvmStatic       // only members in object or companion object annotated with @JvmStatic
    fun main(args: Array<String>) {

        val singletonObject = SingletonObject.getInstance()
        singletonObject.showMessage()

    }
}
